numba/cpython/listobj.py

```diff
@@ -1121,6 +1121,19 @@ def impl(iterable, key=None, reverse=False):
         return lst
     return impl
 
+
+@overload(reversed)
+def list_reverse(A):
+    """
+    reversed for lists and arrays
+    """
+    def impl_reversed(A):
+        A_len = len(A)
+        for i in range(A_len):
+            yield A[A_len - 1 - i]
```

Comment:
If this worked via iterator protocol then something like this could work?
```suggestion
        for i in reversed(range(A_len)):
            yield A[i]
```